docs/usage.rst

```diff
@@ -0,0 +1,719 @@
+
+    backend.date('/file.txt', '1.0.0')
+
+.. jupyter-execute::
+
+    @add_method(SQLite)
+    def _owner(
+            self,
+            path: str,
+            version: str,
+    ) -> str:
+        with self._db as db:
+            query = f'''
+                SELECT owner
+                FROM data
+                WHERE path="{path}" AND version="{version}"
+            '''
+            checksum = db.execute(query).fetchone()[0]
+        return checksum
+
+    backend.owner('/file.txt', '1.0.0')
+
+
+Finally,
+we implement a method
+to fetch a file
+from the backend.
+
+.. jupyter-execute::
+
+    @add_method(SQLite)
+    def _get_file(
+                self,
+                src_path: str,
+                dst_path: str,
+                version: str,
+                verbose: bool,
+        ):
+        with self._db as db:
+            query = f'''
+                SELECT content
+                FROM data
+                WHERE path="{src_path}" AND version="{version}"
+            '''
+            content = db.execute(query).fetchone()[0]
+            with open(dst_path, 'wb') as fp:
+                fp.write(content)
+
+
+Let's verify the file we put on the backend
+contains the expected content.
+
+.. jupyter-execute::
+
+    path = backend.get_file('/file.txt', 'local.txt', '1.0.0')
+    with open(path, 'r') as fp:
+        display(fp.read())
+
+
+To inspect the files
+on our backend
+we provide a listing method.
+
+.. jupyter-execute::
+
+    import typing
+
+    @add_method(SQLite)
+    def _ls(
+            self,
+            path: str,
+    ) -> typing.List[typing.Tuple[str, str]]:
+
+        with self._db as db:
+            if path.endswith('/'):
+                query = f'''
+                    SELECT path, version
+                    FROM data
+                    WHERE path
+                    LIKE ? || "%"
+                '''
+                ls = db.execute(query, [path]).fetchall()
+            else:
+                query = f'''
+                    SELECT path, version
+                    FROM data
+                    WHERE path="{path}"
+                '''
+                ls = db.execute(query).fetchall()
+
+        if not ls and not path == '/':
+            raise FileNotFoundError(
+                errno.ENOENT,
+                os.strerror(errno.ENOENT),
+                path,
+            )
+        return ls
+
+
+Let's test it.
+
+.. jupyter-execute::
+
+    backend.ls('/')
+
+.. jupyter-execute::
+
+    backend.ls('/file.txt')
+
+
+To delete a file
+from our backend
+requires another method.
+
+.. jupyter-execute::
+
+    @add_method(SQLite)
+    def _remove_file(
+            self,
+            path: str,
+            version: str,
+    ):
+        with self._db as db:
+            query = f'''
+                DELETE
+                FROM data
+                WHERE path="{path}" AND version="{version}"
+            '''
+            db.execute(query)
+
+    backend.remove_file('/file.txt', '2.0.0')
```

Comment:
I guess the goal here is to really remove an existing path and not do demonstrate that it will not raise an error when the file does not exist. 
```suggestion
    backend.remove_file('/file.txt', '1.0.0')
```